docs/adding_an_e2e_test.md

```diff
@@ -0,0 +1,106 @@
+# Adding an E2E Test
```

Comment:
nit: can you use a markdown formatter to make this 80 columns?

Reply:
Done!